app/common/model/src/main/java/io/syndesis/common/model/connection/PropertyRelation.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.common.model.connection;
+
+import java.io.Serializable;
+import java.util.List;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+@Value.Immutable
+@JsonDeserialize(builder = PropertyRelation.Builder.class)
+@SuppressWarnings("immutables")
+public interface PropertyRelation extends Serializable {
+
+    class Builder extends ImmutablePropertyRelation.Builder {
+        // make ImmutablePropertyRelation.Builder accessible
+    }
+
+    @Value.Immutable
+    @JsonDeserialize(builder = PropertyRelationWhen.Builder.class)
+    interface PropertyRelationWhen {
```

Comment:
Can we avoid inner interfaces with immutables ? IDEA doesn't work with them. Alternatively, if you find a way how to compile this from within IDEA, the better ;-)

Actually, I have no idea what `PropertyRelationWhen` means. Could you elaborate a bit ? Actually, I would expect some date here, but is probably not.

Reply:
Hmm. not sure about the IDEA thing, but for `Meta` I had to extract an inner interface to make IDEA happy. But I see that there are also other inner interfaces used, so might be not an issue here.

But when keeping it, then I would remove teh `PropertyRelation` prefix, as `PropertyRelation.PropertyRelationWhen` is redundant.

Reply:
I could be mistaken, but I believe PropertyRelationWhen may correspond to the structure found here - https://github.com/udos86/ng-dynamic-forms/#related-form-controls, which is something we need to help out the forms interactivity in the UI. Seeing that may make sense out of some of this :)

Reply:
What @seanforyou23 said about what the properties mean. For the name, I'm ok with renaming the inner classes to remove the redundant PropertyRelation prefix.